Add docs for sso-team-id to the Teams API (#2210)
hashicorp/terraform-website: content/cloud-docs/api-docs/teams.mdx

> -> **Note:** Team management is a paid feature, available as part of the **Team** upgrade package. [Learn more about Terraform Cloud pricing here](https://www.hashicorp.com/products/terraform/pricing).

Reviewer: What if we combined these two notes. Having two notes in a row like this can be visually overwhelming for users. Combining them into one note, especially since they are related (both paid features) I think is an improvement. Thoughts?

```suggestion
-> **Note:** Team management is a paid feature, available as part of the **Team** upgrade package. Single sign-on is also a paid feature, available as part of the **Business** upgrade package. Refer to [Terraform Cloud pricing](https://www.hashicorp.com/products/terraform/pricing) for details.
```

### Sample Response

> The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.

Reviewer 2: I left a comment on [this](https://github.com/hashicorp/terraform-website/pull/2221) PR but will add it here as well. I imagine we'll want to mention here (and elsewhere that similar text has been added) that this is also going to exist in enterprise and potentially also add text that specifies what version the field will exist in there (e.g., `This attribute is also available in Terraform Enterprise as of version 202204-1` or something similar).

Reviewer: +1  But just to check - is the version we're listing already available? If not, I think we should wait until it is available before adding a future-facing statement.

"The `sso-team-id` attribute is only returned of this team's organization is in the Terraform Cloud Business tier or you set it in Terraform Enterprise version 202204-1 and later."

Because you have to have set it right? (I think that's what I read) I just don't want folks to think it'll always be there in TFE if it won't be.

Reviewer 3: > is the version we're listing already available? If not, I think we should wait until it is available before adding a future-facing statement.

It is not! We won't merge any docs changes (this one or #2221 ) until the changes are actually released.

> Because you have to have set it right? (I think that's what I read) I just don't want folks to think it'll always be there in TFE if it won't be.

You do need to set the value, yes. I will double check if the API returns the empty field (`sso-team-id : ""`) if you dont have it set, or if SSO isnt enabled...

> _The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier._

Reviewer: I don't think we need italics!

```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```

      "type": "teams",
      "attributes": {
        "name": "team-creation-test",

>         "sso-team-id": "cb265c8e41bddf3f9926b2cf3d190f0e1627daa4",

Author: This field is only returned if the organization that this team belongs to is in the Business tier. How should I express this?

Reviewer: My instinct is to have one sentence under the Sample Response header that explains this.  Thoughts on this approach? I think it's better to state this in each place where the user could see this and potentially get confused. So each response that has this attribute in it would now have this brief explanatory line. I don't want to put it in a note because that feels like it would get visually very busy if we used it throughout the page. So something like...

### Sample Response

The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.

Reviewer 3: Updated this based on your comment @laurapacilio

| --------------------------------------- | ------ | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `data.type`                             | string |            | Must be `"teams"`.                                                                                                                                                                                                                   |
| `data.attributes.name`                  | string |            | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.                                                                              |

> | `data.attributes.sso-team-id`           | string | (nothing)  | The unique identifier of the team referred to from the SAML MemberOf attribute.                                                                                                                                                      |

Author: This field is optional and is only usable if the team that is being created/updated belongs to an organization that is in the business tier.

| --------------------------------------- | ------ | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `data.type`                             | string |            | Must be `"teams"`.                                                                                                                                                                                                                   |
| `data.attributes.name`                  | string |            | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.                                                                              |

> | `data.attributes.sso-team-id`           | string | (nothing)  | The unique identifier of the team referred to from the SAML MemberOf attribute.                                                                                                                                                      |

Reviewer: ```suggestion
| `data.attributes.sso-team-id`           | string | (nothing)  | The unique identifier of the team from the SAML `MemberOf` attribute. Only available if the team belongs to an organization in the paid Terraform Cloud Business tier.                                                                                                                                                     |
```

    "type": "teams",
    "attributes": {
      "name": "team-creation-test",

>       "sso-team-id": "cb265c8e41bddf3f9926b2cf3d190f0e1627daa4",

Reviewer 3: Just wanted to call out that I **didn't** add a _The sso-team-id attribute is only returned if this team's organization is in the Terraform Cloud Business tier._ note here b/c it is called out in the table immediately above

Reviewer: okay that's fair and makes sense to me!

### Sample Response

> _The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier._

Reviewer: ```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```

| --------------------------------------- | ------ | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `data.type`                             | string |                  | Must be `"teams"`.                                                                                                                                                                                                                   |
| `data.attributes.name`                  | string | (previous value) | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.                                                                              |

> | `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team referred to from the SAML MemberOf attribute.                                                                                                                                                      |

Reviewer: ```suggestion
| `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team from the SAML `MemberOf` attribute.  Only available if the team belongs to an organization in the paid Terraform Cloud Business Tier.                                                                                                                                                   |
```

| --------------------------------------- | ------ | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `data.type`                             | string |                  | Must be `"teams"`.                                                                                                                                                                                                                   |
| `data.attributes.name`                  | string | (previous value) | The name of the team, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.                                                                              |

> | `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team from the SAML `MemberOf` attribute.  Only available if the team belongs to an organization in the paid Terraform Cloud Business Tier.                                                                                                                                                   |

Reviewer: ```suggestion
| `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team from the SAML `MemberOf` attribute. Only available if the team belongs to an organization in the paid Terraform Cloud Business tier.                                                                                                                                                   |
```

### Sample Response

> _The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier._

Reviewer: ```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```
